examples/synthetics_example_test.go

```diff
@@ -143,125 +138,519 @@ func demonstrateSyntheticsTestsAPI() error {
+func newHostnameTest(ctx context.Context, client *kentikapi.Client) (*synthetics.Test, error) {
+	getAllResp, err := client.Synthetics.GetAllAgents(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("client.Synthetics.GetAllAgents: %w", err)
+	}
+
+	agentID, err := pickDualIPRustAgentID(getAllResp.Agents)
+	if err != nil {
+		return nil, fmt.Errorf("pick agent ID for hostname test: %w", err)
+	}
+
+	test := synthetics.NewHostnameTest(synthetics.HostnameTestRequiredFields{
+		BasePingTraceTestRequiredFields: synthetics.BasePingTraceTestRequiredFields{
+			BaseTestRequiredFields: synthetics.BaseTestRequiredFields{
+				Name:     "go-sdk-example-hostname-test",
+				AgentIDs: []string{agentID},
+			},
+			Ping: synthetics.PingSettingsRequiredFields{
+				Timeout:  10 * time.Second,
+				Count:    10,
+				Protocol: synthetics.PingProtocolTCP,
+				Port:     65535,
+			},
+			Traceroute: synthetics.TracerouteSettingsRequiredFields{
+				Timeout:  59999 * time.Millisecond,
+				Count:    5,
+				Delay:    100 * time.Millisecond,
+				Protocol: synthetics.TracerouteProtocolUDP,
+				Limit:    255,
+			},
+		},
+		Definition: synthetics.TestDefinitionHostnameRequiredFields{
+			Target: "www.example.com",
+		},
+	})
+
+	test.Settings.Period = 15 * time.Minute
+	test.Settings.Family = synthetics.IPFamilyV4
+	test.Settings.NotificationChannels = []string{} // must contain IDs of existing notification channels
+	test.Settings.Health = synthetics.HealthSettings{
+		LatencyCritical:           50 * time.Millisecond,
+		LatencyWarning:            20 * time.Millisecond,
+		LatencyCriticalStdDev:     100 * time.Millisecond,
+		LatencyWarningStdDev:      100 * time.Millisecond,
+		JitterCritical:            50 * time.Millisecond,
+		JitterWarning:             20 * time.Millisecond,
+		JitterCriticalStdDev:      100 * time.Millisecond,
+		JitterWarningStdDev:       100 * time.Millisecond,
+		PacketLossCritical:        100,
+		PacketLossWarning:         100,
+		HTTPLatencyCritical:       50 * time.Millisecond,
+		HTTPLatencyWarning:        20 * time.Millisecond,
+		HTTPLatencyCriticalStdDev: 100 * time.Millisecond,
+		HTTPLatencyWarningStdDev:  100 * time.Millisecond,
+		HTTPValidCodes:            []uint32{http.StatusOK, http.StatusCreated},
+		DNSValidCodes:             []uint32{1, 2, 3},
+		UnhealthySubtestThreshold: 2,
+		AlarmActivation: &synthetics.AlarmActivationSettings{
+			TimeWindow:  75 * time.Minute,
+			Times:       4,
+			GracePeriod: 3,
+		},
+	}
+	test.Settings.Ping.Delay = 100 * time.Millisecond
+	test.Settings.Ping.Port = 65535
+	test.Settings.Traceroute.Port = 1
+
+	return test, nil
+}
+
+func createMinimalTests() error {
+	ctx := context.Background()
+	client, err := NewClient()
+	if err != nil {
+		return err
+	}
+
+	getAllResp, err := client.Synthetics.GetAllAgents(ctx)
+	if err != nil {
+		return fmt.Errorf("client.Synthetics.GetAllAgents: %w", err)
+	}
+
+	dualIPRustAgentID, err := pickDualIPRustAgentID(getAllResp.Agents)
```

Comment:
Considering that the goal of the example is to demonstrate the Go SDK (as opposed to test the Kentik API), let's use only IPv4 agents and tests. The reason for this is that the API is little bit behind the development of the Kentik synthetics system, and it currently cannot reliably represent dual stacked agents (it has no place to show the "other" IP address. Which one (v4 or v6) of addresses is shown in the `ip` (and also `local_ip`) property is undefined. The system internally (since about 2 months ago) does the right thing, but it is not visible in the API. The visible behavior may confuse more detailed oriented customers :).

We can add examples of choosing dual stack agents and picking protocol family in the test once the API catches up and we add support for new fields in the SDK.

Reply:
Note that soon (PR is pending review) node.js (`ksynth-agent`) agents will be allowed to run any synthetic test. Rust (`ksynth`) agents will never be able to run `page-load` and `transaction` tests.

Reply:
Modified example to use IPv4 agents: 695420c
Let us know when changes to agents arrive: we'll update docstrings and simplify `TestDemonstrateSyntheticsTestsAPI_CreateMinimalTests`